src/nsntrace.c

```diff
@@ -245,7 +253,9 @@ netns_main(void *arg) {
 	} else if (child_pid > 0) { /* parent - tracer */
 		struct timespec timeout = { 0 };
 
-		waitpid(child_pid, NULL, 0);
+		waitpid(child_pid, &status, 0);
+		ret = WIFEXITED(status) ? WEXITSTATUS(status) : EXIT_FAILURE;
```

Comment:
No strong feelings here. Would you like an updated patch series or are you happy changing it post-merge?